src/state.rs

```diff
@@ -555,13 +564,16 @@ impl ProgressDrawTarget {
         };
     }
 }
+
+const MAX_GROUP_SIZE: f64 = 32.0;
 #[derive(Debug)]
 pub(crate) enum ProgressDrawTargetKind {
     Term {
         term: Term,
         last_line_count: usize,
-        rate: Duration,
+        refresh_rate: Option<u64>,
```

Comment:
It's not clear to me why this is necessary. Can you make this change a separate commit?

Reply:
I added this after the inital pull request as I hit a divide by zero error when disabling the refresh rate.
as I had `ticks = elapsed.as_sec / rate` where `rate` is ms between ticks

So instead of doing 1/rate in the setter I decided to store it as it was passed in as Hz with `None` as no buffering.

I looked a spliting it into two commits but this commit replaces the `last_draw.elapsed() > rate` that which relies on the `let rate = refresh_rate.map(|x| Duration::from_millis(1000 / x))` so they don't really make sense separately.

Reply:
And why can't we keep it `Duration` rather than `u64`?

Reply:
because it is now a frequency measured in Hz (1/s) rather than time meausred in µs

Reply:
I have changed this to Option<LeakyBucket> now to keep all the rate limiting logic in it's own new struct base on your suggestion above.